parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -237,6 +237,36 @@ private static MetadataFilter filter(boolean skipRowGroups) {
   }
 
   /**
+   * read all the footers of the files provided with input streams
+   * (not using summary files)
+   * @param configuration the conf to access the File System
+   * @param partFiles the files to read with equivalent input streams
+   * @param skipRowGroups to skip the rowGroup info
+   * @return the footers
+   * @throws IOException
+   */
+  public static List<Footer> readAllFootersInParallel(final Configuration configuration, Map<FileStatus, FSDataInputStream> partFiles, final boolean skipRowGroups) throws IOException {
```

Comment:
this does not need to be a map (collection of file length and streams?). Also this method does not seem necessary in Parquet. one may do their own threading if necessary when calling readFooter(stream) defined bellow.